contentcuration/contentcuration/serializers.py

```diff
@@ -135,30 +151,39 @@ def update(self, instance, validated_data):
             ThroughModel.objects.bulk_create(bulk_adding_list)
 
+        # print "*********** STARTING: ***********"
+        # recurse(ContentNode.objects.get(id__startswith='f1a03d6'))
         # Perform updates.
         if update_nodes:
-            for node_id, data in update_nodes.items():
-                node = node_mapping.get(node_id, None)
-                if node:
-                    # potential optimization opportunity
-                    for attr, value in data.items():
-                        setattr(node, attr, value)
-                    taglist = []
-                    for tag_data in tag_mapping.get(node_id, None):
-                        # when deleting nodes, tag_data is a dict, but when adding nodes, it's a unicode string
-                        if isinstance(tag_data, unicode):
-                            tag_data = json.loads(tag_data)
-
-                        # this requires optimization
-                        for tag_itm in all_tags:
-                            if tag_itm.tag_name==tag_data['tag_name'] and tag_itm.channel_id==tag_data['channel']:
-                                taglist.append(tag_itm)
-
-                    setattr(node, 'tags', taglist)
-
-                    node.save()
-                    ret.append(node)
-
+            with transaction.atomic():
+                with ContentNode.objects.delay_mptt_updates():
+                    for node_id, data in update_nodes.items():
+                        node = node_mapping.get(node_id, None)
+
+                        if node:
+                            # potential optimization opportunity
+                            for attr, value in data.items():
+                                setattr(node, attr, value)
+                            taglist = []
+                            for tag_data in tag_mapping.get(node_id, None):
+                                # when deleting nodes, tag_data is a dict, but when adding nodes, it's a unicode string
+                                if isinstance(tag_data, unicode):
+                                    tag_data = json.loads(tag_data)
+
+                                # this requires optimization
+                                for tag_itm in all_tags:
+                                    if tag_itm.tag_name==tag_data['tag_name'] and tag_itm.channel_id==tag_data['channel']:
+                                        taglist.append(tag_itm)
+
+                            setattr(node, 'tags', taglist)
+                            node.save()
+                            # if node.parent:
+                            #     ContentNode.objects.move_node(node, ContentNode.objects.get(id=node.parent_id)) # Makes sure cache is updated after save
+                            ret.append(node)
+        # print "*********** FINAL: ***********"
+        # recurse(ContentNode.objects.get(id__startswith='f1a03d6'))
+        clean_db()
```

Comment:
Not sure if this is the best place to put it as there might be other processes that might make the db out of sync. Might be better to just remove this for now